### DISCUSSION TITLE
Components inside .html files o just .jinja files?

### DISCUSSION THREAD

**Discussion by v4ld3rr4m4** [jpsca/jinjax#9] [Q&A] [answered]
How can implement a .jinja component inside a .html file to render it from render_template . ( Yes , I am using flask)

**Comment by jpsca** [comment, marked as answer]
Either if you call `catalog.render(component_name)` from inside a template or from your python code, it's possible to tell JinjaX to search for file with a different extension using the `__file_ext` argument (begins with two underscores).

For example, this code:

```python
catalog.render("Product", __file_ext="html")
```

will search for a `Product.html` file (or for the first file with a name beginning with '_Product_' and ending in '_.html_') **in your components folder(s)**,  instead of the default `Product.jinja`.

Alternatively, you can change the default extension (for all components) when creating the Catalog, with the `file_ext` argument:

```python
catalog = Catalog(file_ext="html")
catalog.add_folder("myapp/components")
```
